### PR TITLE
GT-509 Remove scale_down_candidate annotation

### PR DESCRIPTION
**Pull request by jwierzbo**

### CODE REVIEW

File: pkg/apis/deployment/v1/conditions.go [hunk 1]
```diff
@@ -76,8 +76,6 @@ const (
 	ConditionTypeMemberVolumeUnschedulable ConditionType = "MemberVolumeUnschedulable"
 	// ConditionTypeMarkedToRemove indicates that the member is marked to be removed.
 	ConditionTypeMarkedToRemove ConditionType = "MarkedToRemove"
-	// ConditionTypeScaleDownCandidate indicates that the member will be picked in ScaleDown operaion.
```
Comment (nikita-vanyasin) [inline pkg/apis/deployment/v1/conditions.go:79]: That might be a breaking change for some customer who is using these APIs. I thought we can't do such changes in patch release?

Comment (ajanikow) [reply pkg/apis/deployment/v1/conditions.go:79]: True, it should not be removed - just deprecated

**Review by ajanikow — CHANGES_REQUESTED** [review]
Revert all changes in public api, you can remove only internal parts

File: pkg/apis/deployment/v1/conditions.go [hunk 2]
```diff
@@ -76,7 +76,8 @@ const (
...
+	// ConditionTypeScaleDownCandidate indicates that the member will be picked in ScaleDown operation (Currently not used).
+	// @deprecated will be removed in 1.3.0
```
Comment (nikita-vanyasin) [inline pkg/apis/deployment/v1/conditions.go:80]:
```suggestion
	// Deprecated: will be removed in 1.3.0
```

File: pkg/apis/deployment/v2alpha1/conditions.go [hunk 3]
```diff
@@ -76,7 +76,8 @@ const (
 	ConditionTypeMemberVolumeUnschedulable ConditionType = "MemberVolumeUnschedulable"
 	// ConditionTypeMarkedToRemove indicates that the member is marked to be removed.
 	ConditionTypeMarkedToRemove ConditionType = "MarkedToRemove"
-	// ConditionTypeScaleDownCandidate indicates that the member will be picked in ScaleDown operaion.
+	// ConditionTypeScaleDownCandidate indicates that the member will be picked in ScaleDown operation (Currently not used).
+	// @deprecated will be removed in 1.3.0
```
Comment (nikita-vanyasin) [inline pkg/apis/deployment/v2alpha1/conditions.go:80]:
```suggestion
	// Deprecated: will be removed in 1.3.0
```

**Review by nikita-vanyasin — APPROVED** [review]

**Review by ajanikow — APPROVED** [review]

**Pull request merged**